src/services/userService.js

```diff
@@ -0,0 +1,91 @@
+const { userDao } = require('../models');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const { throwError } = require('../utils/throwError');
+
+const findUser = async (userId) => {
+  return await userDao.findById(userId);
+};
+
+const signUp = async (
+  name,
+  email,
+  password,
+  phoneNumber,
+  zipCode,
+  address,
+  addressDetails
+) => {
+  const emailRegx = /^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
+  if (!email.match(emailRegx)) throwError(400, 'INVALID_EMAIL');
+
+  const passwordRegx = /^(?=.*[0-9])(?=.*[a-z])(?=.*\W)(?!.* ).{8,16}$/;
+  if (!password.match(passwordRegx)) throwError(400, 'INVALID_PASSWORD');
+
+  const saltRounds = 12;
+  const bcryptPassword = await bcrypt.hash(password, saltRounds);
+
+  await userDao.signUp(
+    name,
+    email,
+    bcryptPassword,
+    phoneNumber,
+    zipCode,
+    address,
+    addressDetails
+  );
+};
+
+const duplicateEmail = async (email) => {
+  const existingUser = await userDao.findByEmail(email);
+
+  if (existingUser) throwError(400, 'DUPLICATED_EMAIL_ADDRESS');
+};
+
```

Comment:
```suggestion

const checkDuplicatedEmail = async (email) => {
  const existingUser = await userDao.findByEmail(email);

  if (existingUser) throwError(400, 'DUPLICATED_EMAIL_ADDRESS');
};

```
- 함수명을 기능과 적절하게 선언해주세요.
- duplicateEmail이라는 함수 명은 "이메일을 복제하는 함수"로 오인될 가능성이 있습니다

Reply:
넵 이름 수정 했습니다!